bigbluebutton-html5/imports/api/2.0/users/server/modifiers/changeRole.js

```diff
@@ -0,0 +1,40 @@
+import Logger from '/imports/startup/server/logger';
+import { check } from 'meteor/check';
+import Users from '/imports/api/2.0/users';
+
+export default function handleChangeRole({ body }, meetingId) {
+  const { userId, role, changedBy } = body;
+
+  check(userId, String);
+  check(role, String);
+  check(changedBy, String);
+
+  const selector = {
+    meetingId,
+    userId,
+  };
+
+  const userRoles = [
+    'viewer',
+    role === 'MODERATOR' ? 'moderator' : false,
+  ].filter(Boolean);
```

Comment:
The exact same code is being used in the `addUser` modifier.
https://github.com/bigbluebutton/bigbluebutton/blob/master/bigbluebutton-html5/imports/api/2.0/users/server/modifiers/addUser.js#L64-L68

You probably should make this modifier generic enough to handle any user role modification, then use it on the `addUser` modifier.

Reply:
Same goes for the `handlePresenterAssigned`.